src/libcore/macros/mod.rs

```diff
@@ -687,6 +687,44 @@ macro_rules! todo {
     ($($arg:tt)+) => (panic!("not yet implemented: {}", $crate::format_args!($($arg)+)));
 }
 
+/// Extracts the successful type of a `Poll<T>`.
+///
+/// This macro bakes in propagation of `Pending` signals by returning early.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(future_readiness_fns)]
+/// #![feature(ready_macro)]
+///
+/// use core::task::{Context, Poll};
+/// use core::future::{self, Future};
+/// use core::pin::Pin;
+///
+/// pub fn do_poll(cx: &mut Context<'_>) -> Poll<()> {
+///     let mut f = future::ready(42);
+///     let f = Pin::new(&mut f);
+///
+///     let num = ready!(f.poll(cx));
+///     # drop(num);
+///     // ... use num
+///
+///     Poll::Ready(())
+/// }
+/// ```
```

Comment:
I think it would be beneficial to add here at the bottom (after the example) an indication of what this macro expands to, showing the `match` and `return`. Unlike most std macros, this isn't doing anything particularly magical or clever and seeing the code would help drive home that e.g. it only makes sense to invoke inside of functions that return Poll.